### PR TITLE
Reduce controller server nodes

### PR DESCRIPTION
**Pull request by gezp**
Signed-off-by: zhenpeng ge <zhenpeng.ge@qq.com>

---

## Basic Info

| Info                       | Please fill out this column                                  |
| -------------------------- | ------------------------------------------------------------ |
| Ticket(s) this addresses   | (tickets:  [#816](https://github.com/ros-planning/navigation2/issues/816)) |
| Primary OS tested on       | (Ubuntu20.04, ROS 2 Galactic from binaries)                  |
| Robotic platform tested on | (gazebo simulation of turtlebot3)                            |

-------

## Description of contribution in a few bullet points

As described in [#816](https://github.com/ros-planning/navigation2/issues/816), `rclcpp_node_` in `class ControllerServer ` need be removed, you can find more details(other nodes which need be removed) in [#816 (comment)](https://github.com/ros-planning/navigation2/issues/816#issuecomment-876061677)

after [PR#2459](https://github.com/ros-planning/navigation2/pull/2459) merged, we can create `SimpleActionServer` with argument `spin_thread` which could spin server with dedicated thread.

- set `spin_thread` true when create `action_server_` .

## Test
colcon test
```bash
colcon test --packages-select nav2_controller
#result:0 errors, 0 failures, 0 skipped
```
navigation demo test (turtlebot3 in gazebo)
* success
---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_controller/src/nav2_controller.cpp [hunk 1]
```diff
@@ -266,7 +270,6 @@ ControllerServer::on_cleanup(const rclcpp_lifecycle::State & state)
   odom_sub_.reset();
   vel_publisher_.reset();
   speed_limit_sub_.reset();
-  action_server_.reset();
```
Comment (SteveMacenski) [inline nav2_controller/src/nav2_controller.cpp:269]: Readd this reset, its important for lifecycle bring up and shut downs

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_controller/src/nav2_controller.cpp (hunk 1 above)
Comment (gezp) [reply nav2_controller/src/nav2_controller.cpp:269]: the line265 and line269 are same, i just remove the duplicated one.

File: nav2_controller/src/nav2_controller.cpp [hunk 2]
```diff
@@ -195,8 +195,12 @@ ControllerServer::on_configure(const rclcpp_lifecycle::State & state)
 
   // Create the action server that we implement with our followPath method
   action_server_ = std::make_unique<ActionServer>(
-    rclcpp_node_, "follow_path",
-    std::bind(&ControllerServer::computeControl, this));
+    shared_from_this(),
+    "follow_path",
+    std::bind(&ControllerServer::computeControl, this),
+    nullptr,
+    std::chrono::milliseconds(500),
```
Comment (SteveMacenski) [inline nav2_controller/src/nav2_controller.cpp:202]: I didn't ask in the Planner PR and I should have -- is this 500ms the same behavior as before or did this value change?

Comment (gezp) [reply nav2_controller/src/nav2_controller.cpp:202]: before PR, the default value was used, and 500ms is the default value, so it's same as before.

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**